### PR TITLE
tsdb: return err instead of panic for corrupted chunk

### PR DESCRIPTION
**Pull request by zhulongcheng**
Update https://github.com/prometheus/prometheus/issues/4128

Related to https://github.com/thanos-io/thanos/issues/1467

### CODE REVIEW

File: tsdb/block_test.go [hunk 1]
```diff
@@ -146,8 +187,28 @@ func TestCorruptedChunk(t *testing.T) {
 			test.corrFunc(f)
 			testutil.Ok(t, f.Close())
 
-			_, err = OpenBlock(nil, blockDir, nil)
-			testutil.Equals(t, test.expErr.Error(), err.Error())
+			// check err in open stage
```
Comment (krasi-georgiev) [inline tsdb/block_test.go]: Please follow Golang commentary guidelines.
https://golang.org/doc/effective_go.html#commentary

**Comment by krasi-georgiev** [comment]
ping @zhulongcheng

**Comment by zhulongcheng** [comment]
@krasi-georgiev updated.

**Comment by zhulongcheng** [comment]
will be fixed by #6040

**Pull request closed without merging**